validator/src/main/java/com/amazon/aoc/validators/ECSHealthCheckValidator.java

```diff
@@ -0,0 +1,80 @@
+package com.amazon.aoc.validators;
+
+import com.amazon.aoc.callers.ICaller;
+import com.amazon.aoc.exception.BaseException;
+import com.amazon.aoc.exception.ExceptionCode;
+import com.amazon.aoc.fileconfigs.FileConfig;
+import com.amazon.aoc.helpers.RetryHelper;
+import com.amazon.aoc.models.Context;
+import com.amazon.aoc.models.ValidationConfig;
+import com.amazon.aoc.services.TaskService;
+import com.amazonaws.services.ecs.model.Container;
+import com.amazonaws.services.ecs.model.DescribeTasksResult;
+import com.amazonaws.services.ecs.model.Task;
+import lombok.extern.log4j.Log4j2;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import java.util.concurrent.TimeUnit;
+
+@Log4j2
+public class ECSHealthCheckValidator implements IValidator {
+  // TODO : need to verify default count
+  private static final int DEFAULT_MAX_RETRY_COUNT = 5;
+  private static final int CHECK_INTERVAL_IN_MILLI = 15 * 1000;
+  private Context context;
+  private int retryCount;
+  private FileConfig expectedMetric;
+
+  private TaskService taskService;
+
+  public ECSHealthCheckValidator(TaskService taskService, int retryCount) {
+    this.taskService = taskService;
+    this.retryCount = retryCount;
+  }
+
+  @Override
+  public void init(Context context, ValidationConfig validationConfig,
+                   ICaller caller, FileConfig expectedDataTemplate) throws Exception {
+    this.context = context;
+    this.expectedMetric = expectedDataTemplate;
+  }
+
+  @Override
+  public void validate() throws Exception {
+    log.info("allow sample app load balancer to start");
+    TimeUnit.SECONDS.sleep(15);
```

Comment:
Do other validation methods do this?  It feels wrong.  The validation shouldn't start until all necessary resources are in place.

Reply:
Other validations methods implement lots of retries to wait for resources to initialize.

Reply:
Thanks. I have updated the code. I left the `log.info` and `time.sleep` by mistake in the code. I have removed it now. ECS resources are created before the validator.